File: examples/json.c
#include <stdbool.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct { const char *foo; } T;
int main(void) { T t = {"bar"}; printf("&{%s}\n", t.foo); printf("{\"foo\":\"%s\"}\n", t.foo); return 0; }

File: examples/json.rb
T = Struct.new(:foo) do
  def to_s
    "&{#{foo}}"
  end
end

json = '{"foo":"bar"}'
foo = json[/\"foo\":\"([^\"]+)\"/, 1]
t = T.new(foo)
puts t
puts "{\"foo\":\"#{t.foo}\"}"
